### PR TITLE
Add info about dependencies when deleting collection

### LINKED ISSUE #17536: App: Trying to remove a collection with dependencies throws error which is not clear
### Describe the Bug

When trying to remove a collection which have other tables pointing to it (foreign keys), user receives an error but does not tell much.
Here's an example:
image

Users should know what relations are pointing to this table in order to quickly remove those relations and finally remove the table. For example:
```
This table is still referenced by "hero" field in "Page" collection.
Please remove this field first // Optional
```

This is particularly frustrating for Cloud users since they do not have access to foreign keys.

### To Reproduce

- Create collection A
- Create collection B
- Create M2O relation in B pointing to A
- Try to remove collection A

### Errors Shown

_No response_

### What version of Directus are you using?

9.21.2 (Cloud)

### What version of Node.js are you using?

Cloud

### What database are you using?

Cloud

### What browser are you using?

Chrome

### How are you deploying Directus?

Cloud

### PR DESCRIPTION
**Pull request by Nitwel**
This PR introduces a warning showing the user in advance why the deletion process might fail.

Fixes #17536
Fixes ENG-701

grafik

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
@Nitwel Do we know for a fact that the deletion will fail? Cause if that's the case, I'd rather just say "You can't delete collection this as other items are relying on it. Please delete the other collection first"

**Comment by Nitwel** [comment]
Good point, I went with that wording for now as I had the suspicion that DB vendors like SQLite might not support foreign key constraints.

**Review by br41nslug — COMMENTED** [review]
It's working as advertised but I find the double popup on postgres a bit counter intuitive 😬 It feels a bit redundant to get a "confirmation" popup where you agree and then it doesnt delete it and you have to go back delete it manually anyway.

https://user-images.githubusercontent.com/9389634/228194801-bf15e9b5-dfcc-4ed7-810c-8a8e64cd92e4.mp4

In sqlite it makes a lot more sense as it doesnt get constraint blocked. However both postgres and sqlite have issues with circular dependencies, that may be a separate issue.


https://user-images.githubusercontent.com/9389634/228195919-7d99513b-b1db-47a6-8da9-7ecf376372a5.mp4


https://user-images.githubusercontent.com/9389634/228195937-401e15b2-1ddf-492e-ad88-5d4bc15d4fd4.mp4

**Comment by Nitwel** [comment]
@rijkvanzanten Tim and I discussed that it might make more sense to not only warn the user that you might not be able to delete something but instead to delete the fields that the collection depends on first and then delete the collection, so you don't have to do that yourself manually. Thoughts?

**Comment by Zack-Heisnberg** [comment]
> @rijkvanzanten Tim and I discussed that it might make more sense to not only warn the user that you might not be able to delete something but instead to delete the fields that the collection depends on first and then delete the collection, so you don't have to do that yourself manually. Thoughts?

it's should have a warning though that the data of that field will be lost  

This collection " a " depends on the collection you want to delete " b " , 
do you want to delete the field and all it's data "x"  in items in collection "a" to delete the collection "b"

**Review by paescuj — COMMENTED** [review]
Works as expected for collections with M2O fields 👍 

Not sure about other types of relations, though. For example when deleting a collection included in a M2A relation it now deletes the `item` field in the junction table - this wasn't the case before:

<table><tr><td>

</td></tr></table>

**Comment by rijkvanzanten** [comment]
@br41nslug @paescuj Y'all happy?!!!!!

**Review by rijkvanzanten — APPROVED** [review]
This is one of those PRs that should've been there from the beginning. Thanks @Nitwel!

**Comment by paescuj** [comment]
> @br41nslug @paescuj Y'all happy?!!!!!

Not yet, as outlined in https://github.com/directus/directus/pull/17961#pullrequestreview-1362663323
I really think the `item` field shouldn't get deleted!

**Review by br41nslug — APPROVED** [review]
Looks to be working as expected now on sqlite and postgres 😄

File: app/src/lang/translations/en-US.yaml [hunk 1]
```diff
@@ -679,6 +679,7 @@ one_filtered_item: '1 filtered item'
 delete_collection_are_you_sure: >-
   Are you sure you want to delete the collection "{collection}"? This will delete the collection and all items in it. This action is permanent.
 delete_folder_are_you_sure: Are you sure you want to delete the folder "{folder}"? Nested folders and collections will be moved to the top most level.
+delete_collection_peer_dependencies: The following fields will be deleted as well, as they share a relation with the current collection.
```
Comment (rijkvanzanten) [inline app/src/lang/translations/en-US.yaml:682]:
```suggestion
delete_collection_peer_dependencies: The following fields will be deleted as well, as they are related to the current collection.
```

**Comment by paescuj** [comment]
> @paescuj I updated to only delete the item field in an a2o, when there is only 1 collection left, thus it is fine to delete it. Otherwise it won't be deleted.

Okay, that already makes me much happier 👍 But I still think that there's no need to deleted this field at all. So the M2A field would remain usable in that you can simply connect a new collection (as it is the case now):
Screenshot 2023-04-12 at 17 20 21

**Comment by Nitwel** [comment]
Agreed 👍 
Updated it to only delete m2o relations.

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**